inst/htmljs/animint.js

```diff
@@ -1752,8 +1771,13 @@ var animint = function (to_select, json_file) {
 	});
       }
     }else{//has neither clickSelects nor clickSelects.variable.
-      elements.style("opacity", get_alpha)
-      if(g_info.geom != "text"){
+      elements.style("opacity", get_alpha);
+      // geom_segment/linerange/hline/vline no `stroke` with no clickSelects
+      const excludedGeoms = ["segment", "linerange", "hline", "vline"];
```

Comment:
this logic is strange to me, why not instead enumerate the geoms that do support fill? (rather than the current code which enumerates the geoms which do not support fill)
even better, I wonder if we can define some class/inheritance structure which takes care of this kind of logic, and would avoid this kind of special case code.
GeomPoint inherits from GeomsWhichSupportFill inherits from general Geom etc?

Reply:
I'm a little confused, you mean to create separate classes for each geom type, and within those classes, define the styles and behaviors that are specific to that geom? 
That we should refactor this existing code into a class-based structure?https://github.com/animint/animint2/blob/efa91c4e783569bb236e7f1702ddba7eb0946374/inst/htmljs/animint.js#L1522-L1531

Reply:
right. that would be cleaner / easier to maintain than the current approach using if/then in different places. maybe we can save that for a future PR though if you think it is too complicated?

Reply:
great